src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs

```diff
@@ -0,0 +1,254 @@
+                }
+
+                // Deduct the number of bots that was just spawned from the needed count of bots
+                botsNeeded -= botsToSpawnNow;
+
+                // Update each bot
+                foreach (var bot in bots)
+                {
+                    bot.TickAsync(now, warmup).Wait();
+                }
+
+                // Remove bots that are done
+                bots.RemoveAll(b => b.SessionEnded);
+
+                now += TimeSpan.FromSeconds(step);
+            }
+
+            client.SendMessagesInQueueAsync().Wait();
+        }
+
+        private void GenerateRoutes()
+        {
+            var outputPath = ConsoleEx.ReadLine("Output Directory", DefaultRoutesFolder, (s) => Directory.Exists(s));
+            var outputFilePrefix = ConsoleEx.ReadLine("Output File Prefix", DefaultFilePrefix);
+
+            var minLat = ConsoleEx.ReadLine("Min Latitude", MinLatStockholm);
+            var maxLat = ConsoleEx.ReadLine("Max Latitude", MaxLatStockholm);
+            var minLon = ConsoleEx.ReadLine("Min Longitude", MinLonStockholm);
+            var maxLon = ConsoleEx.ReadLine("Max Longitude", MaxLonStockholm);
+
+            var maxDistance = ConsoleEx.ReadLine("Max Distance (m), 0=unlimited", 0);
+            var maxTotalWalkingTime = ConsoleEx.ReadLine("Max Walking Time (s), 0=unlimited", 0);
+
+            var noRoutes = ConsoleEx.ReadLine("Number of routes to generate", DefaultNumberOfRoutes);
+
+            var rnd = new Random();
+
+
+            if (s_router == null)
+            {
+                var osmPbfPath = ConsoleEx.ReadLine("OpenStreetMap, OSM, PFB-file", DefaultOsmPbfFilePath, (s) => File.Exists(s));
+
+                s_routerDb = new RouterDb();
+                s_router = new Router(s_routerDb);
+
+                using (var stream = File.OpenRead(osmPbfPath))
+                {
+                    Console.WriteLine("Loading Open Street Map Database. This can take a while!");
+                    var loadOsmDataTask = Task.Run(() =>
+                    {
+                        s_routerDb.LoadOsmData(stream, Vehicle.Pedestrian);
+                    });
+
+                    while (!loadOsmDataTask.IsCompleted)
+                    {
+                        Console.Write(".");
+                        Task.Delay(1000).Wait();
+                    }
+                }
+                Console.WriteLine();
+            }
+
+            Console.WriteLine("Generating random routes");
+
+
+            for (int i = 0; i < noRoutes; i++)
+            {
+                GenerateRandomRoute(i, minLat, maxLat, minLon, maxLon, maxDistance, maxTotalWalkingTime, outputPath, outputFilePrefix);
+
+                if (Console.KeyAvailable)
+                {
+                    if (Console.ReadKey().Key == ConsoleKey.Escape)
+                    {
+                        break;
+                    }
+                }
+            }
+
+            //Parallel.For(0, noRoutes, (i, s) =>
```

Comment:
This commented out section actually works and runs faster but gives you somewhat more problem with debugging. That's why I left it there, but did comment it out.